semanticdb/semanticdb3/javacp.md

```diff
@@ -0,0 +1,551 @@
+  * `ENUM` set for enum types
+* Supported accessibilities for `CLASS` symbols are
+  * `PRIVATE`: set for `private` inner classes
+
+**Enum types** [\[84\]][84] are represented by a single symbol with the `CLASS` kind.
+
+```java
+package a;
+
+public enum Coin {
+  PENNY, NICKEL
+}
+```
+
+<table>
+  <tr>
+    <td><b>Definition</b></td>
+    <td width="275px"><b>Symbol</b></td>
+    <td><b>Kind</b></td>
+    <td><b>Signature</b></td>
+  </tr>
+  <tr>
+    <td><code>Coin</code></td>
+    <td><code>a.Coin#</code></td>
+    <td><code>CLASS</code></td>
+    <td><code>ClassInfoType(List(), List(&lt;Enum&lt;Coin&gt;&gt;), List(&lt;PENNY&gt;, &lt;NICKEL&gt;))</code></td>
+  </tr>
+  <tr>
+    <td><code>Coin</code></td>
+    <td><code>a.Coin#PENNY.</code></td>
+    <td><code>FIELD</code></td>
+    <td><code>TypeRef(None, &lt;Coin&gt;), List())</code></td>
+  </tr>
+  <tr>
+    <td><code>Coin</code></td>
+    <td><code>a.Coin#NICKEL.</code></td>
+    <td><code>FIELD</code></td>
+    <td><code>TypeRef(None, &lt;Coin&gt;), List())</code></td>
+  </tr>
+  <tr>
+    <td><code>Coin</code></td>
+    <td><code>a.Coin#values().</code></td>
+    <td><code>METHOD</code></td>
+    <td><code>MethodType(List(), List(), TypeRef(None, &lt;Array&gt;, List(&lt;Coin&gt;)))</code></td>
+  </tr>
+  <tr>
+    <td><code>Coin</code></td>
+    <td><code>a.Coin#valueOf(String).</code></td>
+    <td><code>METHOD</code></td>
+    <td><code>MethodType(List(), List(), TypeRef(None, &lt;Coin&gt;, List()))</code></td>
+  </tr>
+</table>
+
+Notes:
+* Enum types follow the same rules as class declarations.
+* Enum type classes and enum fields must have the property `ENUM` set.
+* Enum fields have the type of their enclosing class.
+* Enum members include the implicitly declared methods [\[86\]][86] `valueOf` and `values`.
+
+**Interface declarations** [\[77\]][77] are represented by a single symbol
+like classes but with the `INTERFACE` kind. Concretely, the differences
+between interface symbols and class symbols are:
+* Interface symbols do not support properties.
+* Interface members without explicit access modifiers have accessibility
+  `PUBLIC` by default instead of `PRIVATE_WITHIN`.
+
+**Method declarations** [\[82\]][82] are represented by a single symbol with
+the `METHOD` kind and one symbol for each type parameter with kind
+`TYPE_PARAMETER` and formal parameter with kind `PARAMETER`.
+Notes:
+* When compiled with the compiler option `-parameters`, the name of method
+  parameters matches their name written in source. Otherwise, parameters have
+  the name `paramN` where `N` is the index of that given parameter starting at
+  index 0.
+
+**Field declarations** [\[83\]][83] are represented by a single symbol with
+the `FIELD` kind.
+
+**Constructor declarations** [\[90\]][90] are represented by a single symbol with
+name `<init>` and the `CONSTRUCTOR` kind. Constructor formal parameters are represented
+the same way as method declaration formal parameters.
```

Comment:
Looks like the sections about methods, fields and constructors are unfinished. For example, they don't include example code, information about supported properties and accessibilities. Is this intended, or you plan to further work on this PR?

Reply:
Added supported accessibilities.

Still missing examples for fields/methods/constructors